test/ci-operator-configresolver-integration/run.sh

```diff
@@ -9,16 +9,43 @@ set -o pipefail
 
 ROOTDIR=$(pwd)
 WORKDIR="$( mktemp -d )"
-trap "rm -rf ${WORKDIR}" EXIT
+trap 'rm -rf ${WORKDIR}' EXIT
 
-cd $WORKDIR
+cd "$WORKDIR"
 # copy registry to tmpdir to allow tester to modify registry
-cp -a $ROOTDIR/test/ci-operator-configresolver-integration/ tests
-ci-operator-configresolver -config tests/configs -log-level debug &
+cp -a "$ROOTDIR"/test/ci-operator-configresolver-integration/ tests
+ci-operator-configresolver -config tests/configs -log-level debug -cycle 2m &
 PID=$!
-if ! timeout 10s bash -c 2>/dev/null -- "until diff tests/expected/openshift-installer-release-4.2.json <(curl 'http://127.0.0.1:8080/config?org=openshift&repo=installer&branch=release-4.2'); do sleep 1; done" &>/dev/null; then
-    echo "diff: $(diff -ru tests/expected/openshift-installer-release-4.2.json <(curl 'http://127.0.0.1:8080/config?org=openshift&repo=installer&branch=release-4.2'))"
+# generation should be >= 1. 0 or an error means the config function has not run yet
+# shellcheck disable=SC2016
+if ! timeout 10s bash -c 2>/dev/null -- 'until [[ $(curl http://127.0.0.1:8080/generation 2>/dev/null) -ne 0 ]]; do sleep 1; done'; then
+    echo "Timed out waiting for generation to equal 1"
     kill $PID
+    wait $PID
+    exit 1
+fi
+if ! diff tests/expected/openshift-installer-release-4.2.json <(curl 'http://127.0.0.1:8080/config?org=openshift&repo=installer&branch=release-4.2' 2>/dev/null) &>/dev/null; then
+    echo "diff: $(diff -ru tests/expected/openshift-installer-release-4.2.json <(curl 'http://127.0.0.1:8080/config?org=openshift&repo=installer&branch=release-4.2' 2>/dev/null))"
+    kill $PID
+    wait $PID
+    exit 1
+fi
+currGen=$(curl 'http://127.0.0.1:8080/generation')
+export currGen
```

Comment:
Why do you need to export this?

Reply:
Oh, because of the child of `timeout`?

Reply:
Yeah. I spent like half an hour trying to get that to work before realizing I needed to export `currGen`.